ra/ra.go

```diff
@@ -90,20 +91,49 @@ func NewRegistrationAuthorityImpl(
 		log:   logger,
 		authorizationLifetime:        DefaultAuthorizationLifetime,
 		pendingAuthorizationLifetime: DefaultPendingAuthorizationLifetime,
-		rlPolicies:                   policies,
-		tiMu:                         new(sync.RWMutex),
-		maxContactsPerReg:            maxContactsPerReg,
-		keyPolicy:                    keyPolicy,
-		maxNames:                     maxNames,
-		forceCNFromSAN:               forceCNFromSAN,
-		regByIPStats:                 scope.NewScope("RA", "RateLimit", "RegistrationsByIP"),
-		pendAuthByRegIDStats:         scope.NewScope("RA", "RateLimit", "PendingAuthorizationsByRegID"),
-		certsForDomainStats:          scope.NewScope("RA", "RateLimit", "CertificatesForDomain"),
-		totalCertsStats:              scope.NewScope("RA", "RateLimit", "TotalCertificates"),
+		rlMu:                 new(sync.RWMutex),
```

Comment:
Rather than making `rlMu` a property of the RA, can you make it a property of the RateLimitConfig object? This probably requires giving RateLimitConfig an additional level of nesting so that it can swap out an internal pointer, and then adding accessors for the different policies. But it means somewhat better encapsulation.